objects/src/accounts/storage.rs

```diff
@@ -0,0 +1,50 @@
+use super::{Digest, Hasher, Word, ZERO};
+
+// TYPE ALIASES
+// ================================================================================================
+
+pub type StorageItem = (Word, Word);
+
+// ACCOUNT STORAGE
+// ================================================================================================
+
+/// Account storage is a key-value map where both keys and values are words (4 elements).
+///
+/// Internally, account storage uses a compact Sparse Merkle tree to store the data. To ensure that
+/// the data is randomly distributed across the tree, the index of item in the tree is derived by
+/// hashing the key.
+#[derive(Debug, Clone, Default, PartialEq, Eq)]
+pub struct AccountStorage {
+    // TODO: add compact SMT as backing storage
+}
+
+impl AccountStorage {
```

Comment:
this also looks like a potential trait instead of a concrete structure. the topic is discussed below

Reply:
Maybe, but I'm not sure about this yet. How exactly this struct will be used in a bigger system is still TBD.